tt_metal/host_api.hpp

```diff
@@ -273,12 +274,12 @@ void EnqueueReadBuffer(CommandQueue& cq, Buffer& buffer, vector<uint32_t>& dst,
  *
  * | Argument     | Description                                                            | Type                          | Valid Range                            | Required |
  * |--------------|------------------------------------------------------------------------|-------------------------------|----------------------------------------|----------|
- * | cq           | The command queue object which dispatches the command to the hardware  | CommandQueue &                |                                        | Yes      |
+ * | cq           | The command queue object which dispatches the command to the hardware  | CommandQueue &              |                                        | Yes      |
  * | buffer       | The device buffer we are reading from                                  | Buffer &                      |                                        | Yes      |
  * | dst          | The memory where the result will be stored                             | void*                         |                                        | Yes      |
  * | blocking     | Whether or not this is a blocking operation                            | bool                          | Only blocking mode supported currently | Yes      |
  */
-void EnqueueReadBuffer(CommandQueue& cq, Buffer& buffer, void* dst, bool blocking);
+void EnqueueReadBuffer(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer> > buffer, void * dst, bool blocking);
```

Comment:
Is adding _size_t transfer_size_ argument going to be added in this PR or later PRs?

Reply:
I'll add that as a follow-on PR when I do the hookups in the queue APIs in eager land